spec/javascripts/selectors/screening/decisionShowSelectorsSpec.js

```
@@ -246,6 +247,38 @@ describe('allegationShowSelectors', () => {
     })
   })
 
+  describe('selectContactReference', () => {
+    it('returns the value of the field', () => {
+      const screening = {screening_contact_reference: '111-222'}
+      const state = fromJS({screening})
+      expect(selectContactReference(state).get('value')).toEqual('111-222')
+    })
+
+    it('returns empty if the value of the field is null', () => {
+      const screening = {screening_contact_reference: null}
+      const state = fromJS({screening})
+      expect(selectContactReference(state).get('value')).toEqual('')
+    })
+
+    it('returns an error if the value is not a valid case or referral id', () => {
+      const screening = {screening_contact_reference: '0442'}
+      const involvements = {
+        referrals: [{
+          start_date: '01/01/2014',
+          legacy_descriptor: {
+            legacy_ui_id: '0442-2654-1834-4001650',
+          },
+        }],
+        cases: [{
+          start_date: '01/01/2014',
+          end_date: '02/02/2014',
+        }],
+      }
+      const state = fromJS({screening, involvements})
+      expect(selectContactReference(state).get('errors')).toEqual(List([]))
```

`it` description suggests this should return an error?

---

+1